torch/_subclasses/meta_utils.py

```diff
@@ -1582,16 +1643,14 @@ def is_c_of_r(
                                 sizes,
                                 strides,
                                 dtype=t.dtype,
-                                device="meta",
+                                device=override_device or t.device,
```

Comment:
I can't tell if this is semantics preserving or not

Reply:
This is on the older version of this PR - there was a collision w/ #141839 (which landed yesterday) which implements similar functionality in a different way - so I ripped the override stuff out and just used that.